update DG

yyingg-243 opened the pull request:
#45
Update DG
Edit sequence diagram ( attack, defend)
Create class diagram for rolladieException and Storage class update class and test structure

yyingg-243 committed:
#45
Update DG
Edit sequence diagram ( attack, defend)
Create class diagram for rolladieException and Storage class
update class and test structure

yyingg-243 committed:
Merge branch 'master' of https://github.com/AY2425S2-CS2113-T13-4/tp

TiangSoonYong commented on docs/uml_image/exceptionClassDiagram.png:
I think we have to remove the class icons as well as represent the correct visibility icons as per the instructions

TiangSoonYong commented on docs/uml_image/storageClass.png:
I have actually done up my Storage class diagram in PR #56, maybe we can discuss to see which to use

yyingg-243 replied:
just have a quick check on both, I think yours is better cause mine was just a draft version (hvt really deal with the details)so ill remove mine on my side.

TiangSoonYong commented on docs/uml_image/attackSequence.png:
It is quite difficult to follow the sequence of the program and there are some errors, for example Parser return inputString to PlayerTurn but PlayerTurn did not call Parser, which should not happen. 

Perhaps we should remove the footboxes and the labelling for the classes should reflect that it is an object. I think the current interpretation in the diagram implies that all the methods are static.

yyingg-243 replied:
done removing footbox and updated the diagram to let player turn class parser before parser returning input string to it .

yyingg-243 committed:
Merge branch 'master' of https://github.com/AY2425S2-CS2113-T13-4/tp

TiangSoonYong reviewed (commented):
The DG is well put together, but the UML diagrams may need some edit!

yyingg-243 committed:
Merge branch 'master' of https://github.com/AY2425S2-CS2113-T13-4/tp

# Conflicts:
#	docs/diagrams/attackSequence.puml
#	docs/diagrams/defendSequence.puml
#	docs/diagrams/lootSequence.puml
#	docs/diagrams/shopSequence.puml

yyingg-243 committed:
update dg and diagram according to member's comment

fix attack and defned diagram to correctly reflect the parser and player turn call
deleted storage class diagram
updated dg components detail

yyingg-243 committed:
update readme file
update ppp

yyingg-243 committed:
#63
resolve naming error: package name should be lower case
edit code according to check style error report

yyingg-243 committed:
update and add architecture diagram into DG
update UG
